app/featuresmodel.cpp

```diff
@@ -359,10 +370,27 @@ void FeaturesModel::setLayer( QgsVectorLayer *newLayer )
     }
 
     emit layerFeaturesCountChanged( layerFeaturesCount() );
+    setupSorting();
```

Comment:
Can we call the `setupSorting()` method from the `FeaturesProxyModel` instead of calling it here? That way we would not need to clear the search expression in other models that should not be sorted.

Reply:
That's a great suggestion! I really didn't like having to override `setupSorting()` for models that don't use it...

Reply:
However, this will not work correctly _if_ we use a`FeaturesProxyModel` with `RelationFeaturesModel` or `RelationReferenceFeaturesModel` that should not be sorted (which we shouldn't do anyway)